extensions/roc-package-web-app-react/app/client/render-to-dom.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable global-require */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Router from 'react-router/lib/Router';
+import match from 'react-router/lib/match';
+
+import { rocConfig } from '../shared/universal-config';
+
+function renderSync({ renderProps, createComponent, node, routerRenderFn }) {
+    const finalComponent = createComponent(
+        <Router
+            {...renderProps}
+            render={routerRenderFn}
+        />
+    );
+
+    ReactDOM.render(finalComponent, node);
+}
+
+function renderAsync({ history, routes, ...rest }) {
+    match({ history, routes }, (error, redirectLocation, renderProps) => {
+        renderSync({
+            ...rest,
+            renderProps,
+        });
+    });
+}
+
+export default function renderToDOM(props) {
```

Comment:
I got confused for a second by calling this `props` here since we are in a React context, this has nothing to do with `this.props`. Maybe call it something like `options` instead?